app/src/test/java/org/simple/clinic/login/pin/LoginPinScreenControllerTest.kt

```diff
@@ -114,6 +131,83 @@ class LoginPinScreenControllerTest {
     }
   }
 
+  @Test
+  fun `when new login flow is enabled, if pin is incorrect and submit is clicked, show an error`() {
+    whenever(passwordHasher.compare(any(), any())).thenReturn(Single.just(PasswordHasher.ComparisonResult.DIFFERENT))
+    whenever(userSession.ongoingLoginEntry()).thenReturn(Single.just(OngoingLoginEntry(userId = UUID.randomUUID())))
+    whenever(userSession.saveOngoingLoginEntry(any())).thenReturn(Completable.complete())
+
+    loginConfigEmitter.onNext(LoginConfig(isOtpLoginFlowEnabled = true))
+    uiEvents.onNext(PinTextChanged("1234"))
+    uiEvents.onNext(PinSubmitClicked())
+
+    verify(screen).showIncorrectPinError()
```

Comment:
Can we also do `verify(screen, never()).somethingThatGetsCalledInTheOtherFlow()`?